controllers/metallb_controller.go

```diff
@@ -91,6 +92,28 @@ func (r *MetalLBReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, err
 	}
 
+	instanceList := &metallbv1beta1.MetalLBList{}
+	err = r.Client.List(context.TODO(), instanceList, &client.ListOptions{})
+	if err != nil {
+		return ctrl.Result{}, errors.Wrap(err, "failed listing all MetalLB instances")
+	}
+
+	if len(instanceList.Items) > 0 {
+		if len(instanceList.Items) > 1 {
+			sort.Slice(instanceList.Items, func(i, j int) bool {
+				return instanceList.Items[j].CreationTimestamp.After(instanceList.Items[i].CreationTimestamp.Time)
+			})
+		}
+		if instanceList.Items[0].Name != req.Name {
```

Comment:
here we are just checking that the name we are getting does nto match the first metallb instance that was created, regardless of its validity.